Packs/Nexthink/Integrations/Nexthink/Nexthink.py

```diff
@@ -0,0 +1,187 @@
+            raw_response=deviceList
+        )
+
+        return dArgs
+    else:
+        return 'Endpoint Not Found'
+
+
+def nexthink_installed_packages(device: None, package: None):
+    data = nexthink_request('GET', SEARCH_COMPLIANCE_PACKAGE_DEVICE)
+
+    if len(data) > 0:
+        for t in data:
+            entries = {}
+            entries['Package Name'] = t['package/name']
+            entries['Package Publisher'] = t['package/publisher']
+            entries['Package Version'] = t['package/version']
+            entryList.append(entries)
+
+        deviceEntry['Device Name'] = data[0]['device/name']
+        deviceEntry['Last Logged On User'] = data[0]['device/last_logged_on_user']
+        deviceEntry['IP Address'] = data[0]['device/ip_addresses'][0]
+        deviceEntry['MAC Address'] = data[0]['device/mac_addresses'][0]
+        deviceList.append(deviceEntry)
+        hr = tableToMarkdown('Installed Packages: ', deviceList) + tableToMarkdown('Packages Details: ', entryList)
+
+        dArgs = CommandResults(
+            outputs_prefix="Nexthink.Package",
+            outputs=deviceList,
+            readable_output=hr,
+            raw_response=deviceList
+        )
+
+        return dArgs
+    else:
+        return 'Package Not Found'
+
+
+def nexthink_compliance_check(device: None, ip: None):
+    if not device:
+        data = nexthink_request('GET', SEARCH_DEVICE_USING_IP)
+    else:
+        data = nexthink_request('GET', SEARCH_DEVICE_USING_DEVICE)
+
+    if len(data) > 0:
+        for t in data:
+            entries = {}
+            entries['Device Antivirus'] = t['antivirus_name']
+            entries['Device Antivirus RTP'] = t['antivirus_rtp']
+            entries['Device Antivirus Updated'] = t['antivirus_up_to_date']
+            entries['Device Antispyware'] = t['antispyware_name']
+            entries['Device Antispyware RTP'] = t['antispyware_rtp']
+            entries['Device Antispyware Updated'] = t['antispyware_up_to_date']
+            entryList.append(entries)
+
+        deviceEntry['Device Name'] = data[0]['name']
+        deviceEntry['Last Logged On User'] = data[0]['last_logged_on_user']
+        deviceEntry['IP Address'] = data[0]['ip_addresses'][0]
+        deviceEntry['MAC Address'] = data[0]['mac_addresses'][0]
+        deviceList.append(deviceEntry)
+
+        hr = tableToMarkdown('Endpoint Details :', deviceList) + tableToMarkdown('Compliance Details: ', entryList)
+        dArgs = CommandResults(
+            outputs_prefix="Nexthink.Compliance",
+            outputs=deviceList,
+            readable_output=hr,
+            raw_response=deviceList
+        )
+
+        return dArgs
+    else:
+        return 'Endpoint Not Found'
+
+
+def main():
+    if demisto.command() == 'test-module':
+        data = nexthink_request('GET', TEST_MODULE)
+        if data:
+            return_results("ok")
+        else:
+            return_results(data)
```

Comment:
In case there is no `data`, you should return_error in order to fail the test-module. Please change it.

Reply:
This is being handle from the **nexthink_request** function